### DISCUSSION TITLE
Request support for array and map literal types

### DISCUSSION THREAD

**Discussion by BubbleGarten** [hyperjumptech/grule-rule-engine#457] [General]
Array and map literal are not supported

I want to define a custom function in my code and one of the params is expected to be an array

<img width="1021" alt="image" src="https://github.com/user-attachments/assets/99857703-33c7-4558-b679-1862e0e88195">

**Comment by tlaraMQ** [comment]
I've done something related, searching if a value is inside an array. To do that I defined a field in the fact to be a slice, and then I used a custom function (func isInArray(value int, array []int) bool) to be able to check It. I'm not sure if that is what you want to do, but if it is in that direction I'd be happy to help
